src/finchlite/codegen/c.py

```diff
@@ -204,12 +211,24 @@ def construct_from_c(fmt, c_obj):
     register_property(t, "construct_from_c", "__attr__", lambda fmt, c_value: c_value)
     register_property(t, "numba_type", "__attr__", lambda t: t)
 
+
+def scalar_to_ctypes_copy(fmt, obj):
+    """
+    This hack is required because it turns out that scalars don't own memory or smth
+    """
+    arr = np.array([obj], dtype=obj.dtype, copy=True)
+    scalar_ctype = np.ctypeslib.as_ctypes_type(obj.dtype)
+    ptr_ctype = ctypes.POINTER(scalar_ctype)
+    return arr.ctypes.data_as(ptr_ctype).contents
+
+
 register_property(
     np.generic,
     "serialize_to_c",
     "__attr__",
-    lambda fmt, obj: np.ctypeslib.as_ctypes(obj),
```

Comment:
Can we have it as `np.ctypeslib.as_ctypes(np.array(obj)),` instead? It looks that 0-d arrays print your example correctly. Then we shouldn't need `scalar_to_ctypes_copy`.